src/App/App.js

```diff
@@ -1,14 +1,25 @@
 import React from 'react';
 import { Routes, Route, Navigate } from 'react-router-dom';
 import WikipediaSearch from '../Search/Search.js';
+import { useAuth0 } from "@auth0/auth0-react";
+import { LoginButton, LogoutButton, NavBarButtons } from '../Main/Main.js';
 import './App.css';
 
 function App() {
+  const { isLoading } = useAuth0();
+
+  if (isLoading) {
```

Comment:
love the use of a conditional to make it so that it will display a loading screen, very well done.